pkg/api/v1alpha1/cluster.go

```diff
@@ -841,6 +842,13 @@ func validateCPUpgradeRolloutStrategy(clusterConfig *Cluster) error {
 			return fmt.Errorf("ControlPlaneConfiguration: maxSurge for control plane must be 0 or 1")
 		}
 	case InPlaceStrategyType:
+		if clusterConfig.Spec.DatacenterRef.Kind == VSphereDatacenterKind {
+			if features.IsActive(features.VSphereInPlaceUpgradeEnabled()) {
+				return nil
+			}
+			return fmt.Errorf("in place upgrades are not supported on vSphere. For dev testing, please set the env variable %s", features.VSphereInPlaceEnvVar)
```

Comment:
I would say let's not even mention the feature flag. We don't want people messing around with it and screwing something up
```suggestion
			return errors.New("in place upgrades are not supported on vSphere")
```

Reply:
good callout! updated